ui/core-react/src/core-react/dialog/SimpleDialog.tsx

```diff
@@ -0,0 +1,245 @@
+  minWidth?: string | number;
+  /** Minimum height that the dialog may be resized to. Displayed in px if value is a number; otherwise, displayed in specified CSS unit. Default: 100px */
+  minHeight?: string | number;
+  /** Maximum width that the dialog may be resized to. Displayed in px if value is a number; otherwise, displayed in specified CSS unit. */
+  maxWidth?: string | number;
+  /** Maximum height that the dialog may be resized to. Displayed in px if value is a number; otherwise, displayed in specified CSS unit. */
+  maxHeight?: string | number;
+
+  /** Custom CSS Style for overlay */
+  backgroundStyle?: React.CSSProperties;
+  /** Custom CSS Style for title */
+  titleStyle?: React.CSSProperties;
+  /** Custom CSS Style for footer */
+  footerStyle?: React.CSSProperties;
+  /** Custom CSS class name for the content */
+  contentClassName?: string;
+  /** Custom CSS Style for the content */
+  contentStyle?: React.CSSProperties;
+}
+
+/** itwinui-react [[Dialog]] wrapper that excludes buggy features and adds features from core-react [[Dialog]].
+ * @internal
+ */
+export class SimpleDialog extends React.Component<SimpleDialogProps> {
+  private _parentDocument = document;
+  public static defaultProps: Partial<SimpleDialogProps> = {
+    minWidth: 300,
+    minHeight: 100,
+    width: "50%",
+    modal: true,
+    hideHeader: false,
+    trapFocus: false,
+  };
+
+  constructor(props: SimpleDialogProps) {
+    super(props);
+  }
+
+  public override componentWillUnmount(): void {
+    this._parentDocument.removeEventListener("keyup", this._handleKeyUp, true);
+  }
+
+  public override componentDidMount(): void {
+    this._parentDocument.addEventListener("keyup", this._handleKeyUp, true);
+  }
+
+  // istanbul ignore next
+  public override render(): JSX.Element {
+    const {
+      opened, title, buttonCluster, onClose, onEscape, onOutsideClick, // eslint-disable-line @typescript-eslint/no-unused-vars
+      minWidth, minHeight, width, height, maxHeight, maxWidth,
+      backgroundStyle, titleStyle, footerStyle, style, contentStyle, contentClassName,
+      className, trapFocus, modal, // eslint-disable-line @typescript-eslint/no-unused-vars
+      hideHeader, ...props } = this.props;
+
+    const containerStyle: React.CSSProperties = {
+      margin: "",
+      width, height,
+    };
+
+    const minMaxStyle: React.CSSProperties = {};
+    minMaxStyle.minWidth = (typeof minWidth === "number") ? `${minWidth}px` : minWidth;
+    minMaxStyle.minHeight = (typeof minHeight === "number") ? `${minHeight}px` : minHeight;
+    if (maxWidth !== undefined)
+      minMaxStyle.maxWidth = (typeof maxWidth === "number") ? `${maxWidth}px` : maxWidth;
+    if (maxHeight !== undefined)
+      minMaxStyle.maxHeight = (typeof maxHeight === "number") ? `${maxHeight}px` : maxHeight;
+
+    const divStyle: React.CSSProperties = {
+      ...backgroundStyle,
+      ...style,
+    };
```

Comment:
I know right :) this is how core-react `Dialog` does this (though it makes no sense) and since I needed to preserve as much of props from it as possible, I just kept it.
Should I test whether or not backgroundStyle was implemented correctly (applied to the backdrop) on the core-react `Dialog` and fix the issue if it wasn't?

Reply:
Given that this is `internal`, only used in `MessageBox`, which do not allow this prop, I guess this is not a big deal for now. Is that component intended to replace the `Dialog` at some point ? (Or is it only to not pollute `MessageBox` in the mean time, in which case there is no need to support any of the props that are not needed by `MessageBox`)